Sources/Controllers/TargetMenu/OATransportStopsBaseController.mm

```diff
@@ -189,7 +191,125 @@ - (void) processTransportStop
+            }
+        }
+    }
+    
+    NSMutableArray<OATransportStop *> *localStops = stopAggregated.localTransportStops;
+    NSMutableArray<OATransportStop *> *nearbyStops = stopAggregated.nearbyTransportStops;
+    if (localStops && localStops.count > 0)
+    {
+        return localStops[0];
+    }
+    else if (nearbyStops && nearbyStops.count > 0)
+    {
+        return nearbyStops[0];
+    }
+    return nil;
+}
+
++ (NSArray<OATransportStop *> *) findTransportStopsAt:(double)lat lon:(double)lon radiusMeters:(int)radiusMeters
+{
+    NSMutableArray<OATransportStop *> *transportStops = [NSMutableArray array];
+    
+    const std::shared_ptr<OsmAnd::TransportStopsInAreaSearch::Criteria>& searchCriteria = std::shared_ptr<OsmAnd::TransportStopsInAreaSearch::Criteria>(new OsmAnd::TransportStopsInAreaSearch::Criteria);
+    const auto& point31 = OsmAnd::Utilities::convertLatLonTo31(OsmAnd::LatLon(lat, lon));
+    searchCriteria->bbox31 = (OsmAnd::AreaI)OsmAnd::Utilities::boundingBox31FromAreaInMeters(radiusMeters, point31);
+    
+    OsmAndAppInstance app = [OsmAndApp instance];
+    const auto& obfsCollection = app.resourcesManager->obfsCollection;
+    const auto search = std::make_shared<const OsmAnd::TransportStopsInAreaSearch>(obfsCollection);
+    search->performSearch(*searchCriteria,
+                          [self, transportStops]
+                          (const OsmAnd::ISearch::Criteria& criteria, const OsmAnd::ISearch::IResultEntry& resultEntry)
+                          {
+                              const auto transportStop = ((OsmAnd::TransportStopsInAreaSearch::ResultEntry&)resultEntry).transportStop;
+                              OATransportStop *stop = [[OATransportStop alloc] init];
+                              stop.stop = transportStop;
+                              [transportStops addObject:stop];
+                          });
+    
+    return transportStops;
+}
+
++ (OATransportStopAggregated *) processTransportStopsForAmenity:(NSArray<OATransportStop *> *)transportStops amenity:(OAPOI *)amenity
+{
+    OATransportStopAggregated *stopAggregated = [[OATransportStopAggregated alloc] init];
+    stopAggregated.amenity = amenity;
+    OsmAnd::LatLon amenityLocation = OsmAnd::LatLon(amenity.latitude, amenity.longitude);
+ 
+    for (OATransportStop *stop in transportStops)
+    {
+        stop.transportStopAggregated = stopAggregated;
+        const auto stopExits = stop.stop->exits;
+        BOOL stopOnSameExitAdded = NO;
+        for (const auto exit : stopExits)
+        {
+            const auto loc = exit->location;
+            if (OsmAnd::Utilities::distance(loc, amenityLocation) < ROUNDING_ERROR)
+            {
+                stopOnSameExitAdded = YES;
+                [stopAggregated addLocalTransportStop:stop];
+                break;
+            }
+            if (!stopOnSameExitAdded && OsmAnd::Utilities::distance(stop.stop->location, amenityLocation)
```

Comment:
Do not `OsmAnd::Utilities::distance(stop.stop->location, amenityLocation)` for each exit

Reply:
In android there is distance calculating too:
https://github.com/osmandapp/Osmand/blob/bef177c5d7a415b747e377b7ff2cda844bb6c05f/OsmAnd/src/net/osmand/plus/mapcontextmenu/controllers/TransportStopController.java#L314

Or do you mean I just should to use any another calculating method? Like that, by example?

```
CLLocation *stopCLLocation = [[CLLocation alloc] initWithLatitude:stop.stop->location.latitude longitude:stop.stop->location.longitude];
CLLocation *amenityCLLLocation = [[CLLocation alloc] initWithLatitude:amenityLocation.latitude longitude:amenityLocation.longitude];
if (!stopOnSameExitAdded && [stopCLLocation distanceFromLocation:amenityCLLLocation] <= SHOW_SUBWAY_STOPS_FROM_ENTRANCES_RADIUS_METERS)
{
    [stopAggregated addNearbyTransportStop:stop];
}
```